builtin/logical/pki/backend.go

```diff
@@ -25,6 +25,11 @@ import (
 )
 
 const (
+	operationPrefixPKI        = "pki"
+	operationPrefixPKIIssuer  = "pki-issuer"
+	operationPrefixPKIIssuers = "pki-issuers"
+	operationPrefixPKIRoot    = "pki-root"
```

Comment:
A question of curiosity... is there a HashiCorp company style guide that rules on the point of defining string constants like these? Personally, I find this coding style obfuscates the values when reading the code, and think it would be clearer to not define these constants, and just write the strings in-line where needed.

I've noticed that the current code of the PKI plugin is indecisive about using them - some occurrences of some API parameters are factored out to constants like this, but it's done for a minority of parameters, and not even consistently for the same parameter names.

Reply:
I don't think we have a formal code style guide at HashiCorp. Personally, I tend to refer to [Dave Chenney's](https://dave.cheney.net/practical-go/presentations/qcon-china.html) or the [Google one](https://google.github.io/styleguide/go/) most of the time. I couldn't find anything pertaining to extracting constants in either of those, but I'd be happy to have a more established guidance on this topic :) In most cases I lean towards inlining string constants for better readability, unless they are referred to more than 3-or-so times in multiple functions/files.

Regarding extracting the prefixes into constants, I figured that these are referred to in multiple files and I did not want to introduce typos. It should also signal to the reader that we are referring to a single entity (PKI). If we decide to rename this plugin to something else, we should update all of these prefixes at the same time. I was also considering just using a single `const operationPrefixPKI = "pki"` and appending the "-issuer", etc. inline but the above approach seemed better.